src/server/models/cacheable_queries/campaign-contact.js

```diff
@@ -494,15 +496,28 @@ const campaignContactCache = {
   ) => {
     // console.log('updateSTATUS', newStatus, contact)
     try {
-      await r
+      const assignmentIds = await r
```

Comment:
This is crucially where we create a notification when a message comes in. In order to avoid any new SQL calls, we need to get the assignmentId returned when we do the campaign-contact update.  Before this moment, during an incoming message, we never know the assignmentId, and we can't cache it because reassignments are too common/easy.

If we do have an assignmentId, then we'll look up the assignment in cache and get the user_id that we should save the cache.  The lookup of user_id from the assignment could, in theory, use the database, but assignments should be cached and stay there during a campaign's lifetime.